p2p/starknet/handlers.go

```diff
@@ -134,16 +136,34 @@ func (h *Handler) onBlockBodiesRequest(req *spec.BlockBodiesRequest) (Stream[pro
 }
 
 func (h *Handler) onEventsRequest(req *spec.EventsRequest) (Stream[proto.Message], error) {
-	// todo: read from bcReader and adapt to p2p type
-	count := uint64(0)
+	it, err := h.newIterator(req.Iteration)
+	if err != nil {
+		return nil, err
+	}
+
 	return func() (proto.Message, bool) {
-		if count > 3 {
+		if it.Valid() {
 			return nil, false
```

Comment:
we need to send a `Fin` message when we are done.

https://github.com/NethermindEth/juno/blob/main/p2p/starknet/p2p/proto/event.proto#L24

Reply:
```
    optional BlockID id = 1; // may not appear if Fin is sent to end the whole response
```

Id is optional. We can leave it `nil`